File: tests/fixtures/corpus-full/typescript/cache.ts
export class CacheEntry<T> {
  readonly expiresAt: number;

  constructor(readonly value: T, ttl: number) {
    this.expiresAt = Date.now() + ttl * 1000;
  }

  isExpired(): boolean {
    return Date.now() > this.expiresAt;
  }
}

export class MemoryCache {
  private store = new Map<string, CacheEntry<unknown>>();

  constructor(private defaultTtl = 300) {}

  get<T>(key: string): T | undefined {
    const entry = this.store.get(key) as CacheEntry<T> | undefined;
    if (!entry || entry.isExpired()) return undefined;
    return entry.value;
  }

  set<T>(key: string, value: T, ttl?: number): void {
    this.store.set(key, new CacheEntry(value, ttl ?? this.defaultTtl));
  }

  delete(key: string): void {
    this.store.delete(key);
  }

  flush(): void {
    this.store.clear();
  }
}

File: tests/fixtures/corpus-full/python/cache.py
import time
import hashlib
from typing import Any, Optional, Dict


class CacheEntry:
    def __init__(self, value: Any, ttl: int) -> None:
        self.value = value
        self.expires_at = time.time() + ttl

    def is_expired(self) -> bool:
        return time.time() > self.expires_at


class MemoryCache:
    def __init__(self, default_ttl: int = 300) -> None:
        self.default_ttl = default_ttl
        self._store: Dict[str, CacheEntry] = {}

    def get(self, key: str) -> Optional[Any]:
        entry = self._store.get(key)
        if entry is None or entry.is_expired():
            return None
        return entry.value

    def set(self, key: str, value: Any, ttl: Optional[int] = None) -> None:
        self._store[key] = CacheEntry(value, ttl or self.default_ttl)

    def delete(self, key: str) -> None:
        self._store.pop(key, None)

    def flush(self) -> None:
        self._store.clear()

    def make_key(self, *parts: str) -> str:
        combined = ":".join(parts)
        return hashlib.md5(combined.encode()).hexdigest()
